includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -388,18 +393,19 @@ private function maybe_add_lightbox_attributes( $attributes, $node ) {
 
 		if ( isset( $parent_attributes['data-amp-lightbox'] ) && true === filter_var( $parent_attributes['data-amp-lightbox'], FILTER_VALIDATE_BOOLEAN ) ) {
 			$attributes['data-amp-lightbox'] = '';
-			$attributes['on']                = 'tap:' . self::AMP_IMAGE_LIGHTBOX_ID;
-			$attributes['role']              = 'button';
-			$attributes['tabindex']          = 0;
+			$attributes['lightbox']          = '';
 
-			$this->maybe_add_amp_image_lightbox_node();
+			if ( $is_node_wrapped_in_anchor ) {
+				// Remove the <a> that the image is wrapped in, as it can prevent the lightbox from working.
+				$node->parentNode->parentNode->replaceChild( $node, $node->parentNode );
+			}
```

Comment:
If `amphtml` later supports wrapping `<amp-img lightbox>` in `<a>`, this can be removed.